src/Microsoft.AspNet.OData.Shared/Formatter/Deserialization/ODataResourceSetDeserializer.cs

```diff
@@ -151,10 +196,38 @@ public virtual IEnumerable ReadResourceSet(ODataResourceSetWrapper resourceSet,
                     Error.Format(SRResources.TypeCannotBeDeserialized, elementType.FullName()));
             }
 
-            foreach (ODataResourceWrapper resourceWrapper in resourceSet.Resources)
+            //Ideally we don't need to special case ResourceSetType.ResourceSet, since the code that handles a deltaresourcesetwrapper will also handle a ResourceSetWrapper, 
```

Comment:
```suggestion
            //Ideally we don't need to special case ResourceSetType.ResourceSet, since the code that handles a DeltaResourceSetWrapper will also handle a ResourceSetWrapper, 
```